### ISSUE TITLE
Cannot read property 'toLowerCase' of undefined

### ISSUE THREAD

**Issue by SuperSandro2000** [SchoofsKelvin/vscode-sshfs#122]
````
Error while connecting to SSH FS winsshterm: Cannot read property 'toLowerCase' of undefined
````

````
    "sshfs.configs": [
        {
            "agent": true,
            "host": "example.com",
            "name": "test",
            "privateKeyPath": "some.ppk",
            "putty": true,
            "root": "/",
            "username": "some"
        }
    ]
````

**Comment by SchoofsKelvin** [comment]
Do you have any other configs? Especially ones without a `name` field? It's either a bug in loading configs, or loading PuTTY sessions, so knowing if there are any "incomplete" configurations would be helpful.

**Comment by SuperSandro2000** [comment]
> Do you have any other configs? Especially ones without a name field?

No.

````
{
	"resource": "/c:/Users/name/AppData/Roaming/Code/User/settings.json",
	"owner": "_generated_diagnostic_collection_name_#2",
	"severity": 4,
	"message": "Matches multiple schemas when only one must validate.",
	"startLineNumber": 16,
	"startColumn": 9,
	"endLineNumber": 16,
	"endColumn": 10
}
````

and I have a green line under the first brace in my config example.

**Comment by kurtjcu** [comment]
I am having the same problem, my configs were working until the latest update to vscode...

**Comment by SchoofsKelvin** [comment]
I've tried reproducing this, but can't find a way to trigger this error. I checked every occurrence of `toLowerCase` and every change since 1.14.0, but still can't find the cause.

@SuperSandro2000 One thing I noticed is that you're doing `"agent": true`, which basically gets ignored as it isn't a string. It should either be `pageant` when you're on Windows and want to use Pageant, otherwise (an environment variable resulting in) an absolute path to a (cygwin) UNIX, or something along `/./pipe/...` on Windows. Right now, since you specify `privateKeyPath`, it'll just use that instead.

@kurtjcu Do you know whether only vscode updated, or also this extension? And possibly which version you upgraded from?

I still haven't seen the output from `Output > ssh-fs`, which would help me a lot pinpointing the place where, why and how this error happens.

**Comment by SchoofsKelvin** [comment]
Make sure you have version 1.14.6 of this extension installed before giving me the output. I improved logging a bit, most noticeable explicitly logging an error's stacktrace, as I noticed Error.toString() tends to not do that.

**Comment by SuperSandro2000** [comment]
I installed the extension on a different PC and did not get any errors. Could it be that I tried to run Putty/Pageagent/key from a usb?

**Comment by SchoofsKelvin** [comment]
> Could it be that I tried to run Putty/Pageagent/key from a usb?

I'm not entirely sure that matters, but on the other hand, it shouldn't matter. Only thing I can think of is if you have configured a session with a keypath to some USB, which could possibly be an issue. But pageant-wise it should work, as the extension directly communicates with pageant, not interacting with the USB.

You could retry on the affected pc, but this time first copy the key to your hard disk.

**Comment by clytras** [comment]
I'm having the same issue.

VSCode v1.33.0
SSH FS v1.14.6

The `sshfs.configs`:

```
"sshfs.configs": [
  {
    "host": "site.example.com",
    "name": "site.example.com",
    "password": "password",
    "port": 40022,
    "putty": true,
    "root": "/home/client/public_html/site",
    "username": "client"
  }
]
```

SSH FS wasn't installed at all and wanted to try it today.

![image](https://user-images.githubusercontent.com/6103077/56043369-cb830a00-5d45-11e9-84f3-eb4408ad8018.png)

**Comment by SchoofsKelvin** [comment]
@clytras Could you post the output from `Output > ssh-fs`? You can replace the real data *(e.g. hostname)* with fake data, as you did with your config.

**Comment by clytras** [comment]
Sure, here are the errors:

```
[ERROR] Error while connecting to SSH FS 5shoes.nekya.com:
Cannot read property 'toLowerCase' of undefined
[ERROR] Cannot read property 'toLowerCase' of undefined
TypeError: Cannot read property 'toLowerCase' of undefined
	at exports.modules.54.sessions.filter.s (c:\Users\user\.vscode\extensions\kelvin.vscode-sshfs-1.14.6\dist\5.extension.js:69:55)
	at Array.filter (<anonymous>)
	at Module.<anonymous> (c:\Users\user\.vscode\extensions\kelvin.vscode-sshfs-1.14.6\dist\5.extension.js:69:32)
	at Generator.next (<anonymous>)
	at fulfilled (c:\Users\user\.vscode\extensions\kelvin.vscode-sshfs-1.14.6\dist\5.extension.js:17:58)
```

and here is the complete SSH FS output in a [pastebin](https://pastebin.com/zRzENVhA).

**Comment by SchoofsKelvin** [comment]
Got it, seems like this bug is caused by having PuTTY sessions with an empty hostname. I'm not sure why you would have sessions without hostname (nor did I know that was possible), but I'll add checks for that.

**Comment by clytras** [comment]
@SchoofsKelvin currently I'm having many `KiTTY` sessions inside `WinSSHTerm`, one of it's the same which  SSH-FS throws the error. I also have an SSH tunnel to a macOS client and a couple of sessions with VSCode Remote Workspace plugin, so yes, my case may be an exception. I'm not sure though why there is a session without a hostname.

**Comment by SchoofsKelvin** [comment]
For PuTTY, I directly read all the session data from the Windows Registry. As far as I know, only PuTTY should edit the sessions under `HKEY_CURRENT_USER\Software\SimonTatham\PuTTY\Sessions\`.

I've fixed this bug (b79e478), although if you actually try to use a PuTTY session without hostname (and you don't specify one yourself), it will, of course, fail. This fix is now live in version 1.14.7 of the extension.
